src/components/GameModal/GameModal.test.tsx

```diff
@@ -0,0 +1,55 @@
+import {
+  fireEvent,
+  render,
+  screen,
+} from '@testing-library/react';
+import GameModal from './index';
+
+describe('GameModal', () => {
+  const onClick = jest.fn();
+  const buttons = [{
+    label: 'Button',
+    onClick,
+  }];
+
+  afterEach(() => {
+    onClick.mockReset();
+  });
+
+  it('checking title', () => {
+    render(
+      <GameModal
+        buttons={buttons}
+        title="Title"
+      />,
+    );
+
+    expect(screen.getByText('Title')).not.toBeNull();
+  });
+
+  it('checking button', () => {
+    render(
+      <GameModal
+        buttons={buttons}
+        title="Title"
+      />,
+    );
+
+    fireEvent.click(screen.getByText('Button'));
+
+    expect(onClick).toBeCalled();
```

Comment:
Неплохо бы проверить, что `onClick` вызвался только один раз
```suggestion
    expect(onClick).toBeCalledTimes(1);
```

Reply:
Поправил.